File: poshub-spa/src/app/dtos/updateOrderEventRequestDto.ts
import { ChargeDto } from "./chargeDto";
import { CustomerDto } from "./customerDto";
import { DeliveryDto } from "./deliveryDto";
import { DiscountDto } from "./discountDto";
import { DriverDto } from "./driverDto";
import { ItemDto } from "./itemDto";
import { PaymentDto } from "./paymentDto";
import { TaxDto } from "./taxDto";

export class UpdateOrderEventRequestDto {
  constructor(
    public notes: string = '',
    public orderNumber: string = '',
    public sourceDeviceType: string = '',
    public timeZone: string = '',
    public estimatedDeliveryTime: string = '',
    public payments: PaymentDto[] = [],
    public subTotal: number = 0,
    public driverStatus: string = '',
    public fulfillmentType: string = '',
    public tableName: string = '',
    public totalTax: string = '',
    public total: string = '',
    public discounts: DiscountDto[] = [],
    public currency: string = '',
    public estimatedPickupTime: string = '',
    public delivery: DeliveryDto = new DeliveryDto(),
    public cancellationReason: string = '',
    public tax: TaxDto[] = [],
    public friendlyId: string = '',
    public placedOn: string = '',
    public isPaid: boolean = false,
    public charges: ChargeDto[] = [],
    public driver: DriverDto = new DriverDto(),
    public isScheduledOrder: boolean = false,
    public tableId: number = 0,
    public partnerId: string = '',
    public sourceName: string = '',
    public items: ItemDto[] = [],
    public customer: CustomerDto = new CustomerDto(),
    public status: string = '',
  ) { }
}

File: PosHubApi/Dtos/UpdateOrderEventRequestDto.cs

    using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace PosHubApi.Dtos
{
    public class UpdateOrderEventRequestDto
    {
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string Notes { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string OrderNumber { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string SourceDeviceType { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string Timezone { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public DateTime? EstimatedDeliveryTime { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<PaymentDto> Payments { get; set; } = new();

        public decimal SubTotal { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string DriverStatus { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string FulfillmentType { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string TableName { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public decimal? TotalTax { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public decimal? Total { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<DiscountDto> Discounts { get; set; } = new();

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string Currency { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public DateTime? EstimatedPickupTime { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public DeliveryDto Delivery { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string CancellationReason { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<TaxDto> Tax { get; set; } = new();

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string FriendlyId { get; set; }

        public DateTime PlacedOn { get; set; }

        public bool IsPaid { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<ChargeDto> Charges { get; set; } = new();

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public DriverDto Driver { get; set; }

        public bool IsScheduledOrder { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public int? TableId { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string PartnerId { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string SourceName { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<ItemDto> Items { get; set; } = new();

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public CustomerDto Customer { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string Status { get; set; }
    }

}